stacks-signer/src/v0/signer.rs

```diff
@@ -1405,8 +1511,31 @@ impl Signer {
 
     /// Submit a block for validation, and mark it as pending if the node
     /// is busy with a previous request.
-    fn submit_block_for_validation(&mut self, stacks_client: &StacksClient, block: &NakamotoBlock) {
+    fn submit_block_for_validation(
+        &mut self,
+        stacks_client: &StacksClient,
+        block: &NakamotoBlock,
+        added_epoch_time: u64,
+    ) {
         let signer_signature_hash = block.header.signer_signature_hash();
+        if !self.maybe_processed_parent(stacks_client, block) {
+            let time_elapsed = get_epoch_time_secs().saturating_sub(added_epoch_time);
+            if Duration::from_secs(time_elapsed)
+                < self.proposal_config.proposal_wait_for_parent_time
+            {
+                info!("{self}: Have not processed parent of block proposal yet, inserting pending block validation and will try again later";
+                        "signer_signature_hash" => %signer_signature_hash,
+                );
+                self.signer_db
+                    .insert_pending_block_validation(&signer_signature_hash, added_epoch_time)
+                    .unwrap_or_else(|e| {
+                        warn!("{self}: Failed to insert pending block validation: {e:?}")
+                    });
+                return;
+            } else {
+                debug!("{self}: Cannot confirm that we have processed parent, but we've waiting proposal_wait_for_parent_time, will submit proposal");
```

Comment:
"waited"

Reply:
Fixed in [ac370c0](https://github.com/stacks-network/stacks-core/pull/6030/commits/ac370c06ed04b318dc754677e01190539485a15b)